packages/earth-admin/src/utils/hooks.tsx

```diff
@@ -84,3 +84,20 @@ export function useRequest(
     data,
   };
 }
+
+export function useDomWatcher(ref, callback, skip) {
+  useEffect(() => {
+    if (skip) return;
+
+    const handleClickOutside = event => {
+      if (ref.current && !ref.current.contains(event.target)) {
+        callback && callback();
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [ref, skip]);
+}
```

Comment:
If it's the same hook from map, we could move it into `earth-components` package